src/main/java/org/anarres/parallelgzip/ParallelGZIPOutputStream.java

```diff
@@ -44,25 +46,30 @@ private static DeflaterOutputStream newDeflaterOutputStream(@Nonnull OutputStrea
         return new DeflaterOutputStream(out, deflater, 512, true);
     }
 
-    private static class Block implements Callable<byte[]> {
+    private static class State {
+        private final Deflater def = newDeflater();
+        private final ByteArrayOutputStreamExposed buf = new ByteArrayOutputStreamExposed(SIZE);
+        private final DeflaterOutputStream str = newDeflaterOutputStream(buf, def);
+    }
 
-        private static class State {
+    /** This ThreadLocal avoids the recycling of a lot of memory, causing lumpy performance. */
+    private static final ThreadLocal<State> STATE = ThreadLocal.withInitial(State::new);
```

Comment:
kk, I'm fixing this. The lowest I have installed is 1.7 now (and installing deprecated 1.6 is a real pain these days - so if you want to support that you'll need to test yourself).